### PR TITLE
New bucket features

### PR DESCRIPTION
**Pull request by FerrariDG**
## :pencil: Changelog:

- :sparkles: Features:
  - S3 Bucket funcionalities
    - Check if a bucket exists
    - Create a new bucket
    - Delete an existing bucket
    - List all buckets

### CODE REVIEW

File: tests/unit/buckets/test_create_bucket.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    def test_create_bucket_invalid_name(self, s3_client):
+
+        with pytest.raises(ParamValidationError):
+            create_bucket('')
+
+    def test_create_duplicated_bucket(self, s3_client):
+        create_bucket(BUCKET_NAME)
+
+        response = create_bucket(BUCKET_NAME)
```
Comment (tomassosorio) [inline tests/unit/buckets/test_create_bucket.py:25]: shouldn't throw an error? Or just ignores since already exist? 🤔

**Review by tomassosorio — APPROVED** [review]

File: tests/unit/buckets/test_create_bucket.py (hunk 1 above)
Comment (FerrariDG) [reply tests/unit/buckets/test_create_bucket.py:25]: So, this is a weird behaviour from boto3. When you create a bucket with the same name, it just returns 200 as the first time. This makes it impossible to know what happened on the AWS side 😞

File: tests/unit/buckets/test_create_bucket.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+"""Unit tests for create bucket."""
+from s3_tools import create_bucket
+
+from tests.unit.conftest import BUCKET_NAME
+
+import pytest
+from botocore.exceptions import ParamValidationError
```
[the remark this answers is not in the document]
Comment (FerrariDG) [reply tests/unit/buckets/test_create_bucket.py:7]: I didn't understand. The package deals with the minimum raises possible and let the user deal with them. 
My idea was to show how AWS boto3 validates the name of the bucket, and for some reason, it raises a ParamValidationError even before calling the create function on boot3.

[the remark this answers is not in the document]
Comment (tomassosorio) [reply tests/unit/buckets/test_create_bucket.py:7]: 🤔  if  I run the test like these they pass
```
import pytest
from botocore.exceptions import ParamValidationError

from s3_tools import (
    create_bucket,
    delete_bucket,
)

from tests.unit.conftest import BUCKET_NAME


```

File: s3_tools/buckets/delete.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    session = boto3.session.Session(**aws_auth)
+    s3 = session.client("s3")
+
+    try:
+        response = s3.delete_bucket(Bucket=name)
+    except Exception as error:
+        if isinstance(error, ClientError) and (error.response["Error"]["Code"] == "NoSuchBucket"):
+            return False
+        else:
+            raise error
```
Comment (cmalves) [inline s3_tools/buckets/delete.py:39]: We're missing the `Raises` section in docstring

**Review by cmalves — APPROVED** [review]
✅

**Pull request merged**